presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -1190,4 +1192,17 @@ public HiveClientConfig setS3SelectPushdownMaxConnections(int s3SelectPushdownMa
         this.s3SelectPushdownMaxConnections = s3SelectPushdownMaxConnections;
         return this;
     }
+
+    @Config("hive.temporary-staging-directory-enabled")
+    @ConfigDescription("Should use (if possible) temporary staging directory for write operations")
```

Comment:
`Use temporary staging directory for write operations`. Why do we say `if possible` here?

Reply:
@nezihyigitbasi : For filesystems like S3, we cannot use staging directory since rename is expensive.

Reply:
@nezihyigitbasi Does the above comment answers your question?